### DISCUSSION TITLE
How to handle Email Verification for a flutter mobile app?

### DISCUSSION THREAD

**Discussion by Shiba-Kar** [appwrite/appwrite#4553] [Q&A] [answered]
### 👟 Reproduction steps

1. Create account using email/password
2. Login to account using the same email/password
3. send Email verification request using ``` account.createVerification(url: url); ```
4. Open the link sent to the email
5. The app does not open instead opens the browser. 

### 👍 Expected behavior

The link should navigate to app. 

### 👎 Actual Behavior

The browser is opened.

### 🎲 Appwrite version

Version 1.0.x

### 💻 Operating system

Linux

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by stnguyen90** [comment]
@Shiba-Kar, thanks for raising this issue! 🙏 What did you put for the `url` passed to `account.createVerification()`? Did you configure deep linking for your app so that it opens the app when the URL is opened?

**Comment by Shiba-Kar** [comment]
@stnguyen90 I have configured deeplink . The deeplink works perfectly for OAuth

``` account.createVerification(url:"https://example.com") ```

```xml  
      <activity android:name="com.linusu.flutter_web_auth_2.CallbackActivity" android:exported="true">
            <intent-filter android:label="flutter_web_auth_2">
                <action android:name="android.intent.action.VIEW" />
                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />
                <data android:scheme="appwrite-callback-[PROJECT_ID]" />
            </intent-filter>
        </activity> 
```

**Comment by lohanidamodar** [comment]
> @stnguyen90 I have configured deeplink . The deeplink works perfectly for OAuth
> 
> `account.createVerification(url:"https://example.com")`
> 
> ```
>       <activity android:name="com.linusu.flutter_web_auth_2.CallbackActivity" android:exported="true">
>             <intent-filter android:label="flutter_web_auth_2">
>                 <action android:name="android.intent.action.VIEW" />
>                 <category android:name="android.intent.category.DEFAULT" />
>                 <category android:name="android.intent.category.BROWSABLE" />
>                 <data android:scheme="appwrite-callback-[PROJECT_ID]" />
>             </intent-filter>
>         </activity> 
> ```

During the email verification, you need to handle the redirection yourself from your web app that verifies the email, are you doing that?

**Reply by Shiba-Kar** [reply]
Ya I got it now

**Reply by GG22A** [reply]
Dears, could you please share a full example code, i am newbie and got lost a bit in email verification having exactly the same issue described above in the first message

**Comment by stnguyen90** [comment, marked as answer]
@Shiba-Kar, if you want your mobile app to handle the verification, you'll need to enable deep linking with a URL. See https://docs.flutter.dev/development/ui/navigation/deep-linking#enable-deep-linking-on-android

Once redirected back to your app, you can grab the secret to complete the verification.

**Comment by Shiba-Kar** [comment]
@stnguyen90 is this in the road map to do in the future by default.

**Comment by stnguyen90** [comment]
@Shiba-Kar probably not because we can't modify your manifest file for you and there are steps you have to do manually.
